src/dna_string.rs

```diff
@@ -450,14 +450,37 @@ impl<'a> IntoIterator for &'a DnaString {
 
 
 /// An immutable slice into a DnaString
-#[derive(Eq, PartialEq, Clone)]
+#[derive(Clone)]
 pub struct DnaStringSlice<'a> {
     pub dna_string: &'a DnaString,
     pub start: usize,
     pub length: usize,
     pub is_rc: bool,
 }
 
+impl<'a> PartialEq for DnaStringSlice<'a> {
+    fn eq( &self, other: &DnaStringSlice ) -> bool {
+        let n = self.length;
+        if other.length != n { return false; }
+        for i in 0..n {
+            if self.is_rc == other.is_rc {
+                if self.dna_string.get( self.start + i ) 
```

Comment:
You should be able to use self.get(i) != other.get(i) and let that method deal with the offsets and RC'ing.  It's implemented below as part of the `impl Mer for DnaStringSlice` block.